File: DataBaseRoomApp/jroomapp/src/main/java/com/woowrale/jroomapp/di/components/MainComponent.java
package com.woowrale.jroomapp.di.components;

import com.woowrale.jroomapp.di.modules.MainModule;
import com.woowrale.jroomapp.ui.base.BaseActivity;
import com.woowrale.jroomapp.ui.search.home.SearchActivity;
import com.woowrale.jroomapp.ui.search.local.LocalSearchActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {MainModule.class})
public interface MainComponent {

    void inject(BaseActivity activity);

    void inject(SearchActivity activity);

    void inject(LocalSearchActivity activity);

}

File: DataBaseRoomApp/kroomapp/src/main/java/com/woowrale/kroomapp/di/components/MainComponent.kt
package com.woowrale.kroomapp.di.components

import com.woowrale.kroomapp.di.modules.MainModule
import com.woowrale.kroomapp.ui.base.BaseActivity
import com.woowrale.kroomapp.ui.search.home.SearchActivity
import com.woowrale.kroomapp.ui.search.local.LocalSearchActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [MainModule::class])
interface MainComponent {

    fun inject(activity: BaseActivity)

    fun inject(activity: SearchActivity)

    fun inject(activity: LocalSearchActivity)

}